django/utils/functional.py

```diff
@@ -75,6 +75,10 @@ def __reduce__(self):
                 (func, self.__args, self.__kw) + resultclasses
             )
 
+        def __repr__(self):
+            """ Make sure the lazy object's display is useful when debugging """
```

Comment:
The test failed in PY3 after adding the `if six.PY2:`, after the repr changes mentioned below, so I'd leave it as it is for now.